x/programs/cmd/simulator/src/lib.rs

```diff
@@ -186,7 +186,15 @@ pub struct PlanResponse {
     /// The result of the plan.
     pub result: PlanResult,
     /// An optional error message.
-    pub error: Option<String>,
+    pub error: Option<PlanError>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct PlanError(String);
+impl core::fmt::Debug for PlanError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
```

Comment:
Do you need this? 

It doesn't appear to be doing anything

Reply:
yeah, without it the default behavior is to escape new lines.

`---- tests::mint_and_transfer stdout ----
thread 'tests::mint_and_transfer' panicked at x/programs/rust/examples/token/src/lib.rs:448:9:
error: Some("program execution failed: \n, err: error while executing at wasm backtrace:\n    0: 0x32619 - <unknown>!std::thread::local::LocalKey<T>::try_with::h217d06e356708221\n    1: 0x3253a - <unknown>!std::thread::local::LocalKey<T>::with::h505f088a73c802e5\n    2: 0x32cde - <unknown>!std::thread::local::LocalKey<core::cell::RefCell<T>>::with_borrow_mut::h42f856ca1bb87954\n    3: 0x318e9 - <unknown>!alloc\nnote: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information\n\nCaused by:\n    wasm trap: all fuel consumed by WebAssembly")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace` vs

`---- tests::mint_and_transfer stdout ----
thread 'tests::mint_and_transfer' panicked at x/programs/rust/examples/token/src/lib.rs:448:9:
error: Some(program execution failed:
, err: error while executing at wasm backtrace:
    0: 0x32619 - <unknown>!std::thread::local::LocalKey<T>::try_with::h217d06e356708221
    1: 0x3253a - <unknown>!std::thread::local::LocalKey<T>::with::h505f088a73c802e5
    2: 0x32cde - <unknown>!std::thread::local::LocalKey<core::cell::RefCell<T>>::with_borrow_mut::h42f856ca1bb87954
    3: 0x318e9 - <unknown>!alloc
note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information

Caused by:
    wasm trap: all fuel consumed by WebAssembly)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace`

Reply:
<img width="843" alt="Screenshot 2024-05-29 at 12 28 20 PM" src="https://github.com/ava-labs/hypersdk/assets/91908103/dbb902c0-db78-458f-817c-f80e7fdef872">
<img width="1642" alt="Screenshot 2024-05-29 at 12 29 03 PM" src="https://github.com/ava-labs/hypersdk/assets/91908103/8541a5cc-7aa5-4520-aad4-9e9b073a9fcd">

Reply:
Oh... I see what's happening, it's `Display` formatting instead of debug formatting